testing/matrix/linter/rules/rules.go

```diff
@@ -177,13 +197,35 @@ func containerImageTagLatest(object storage.StoreObject, containers []v1.Contain
 				"Can't parse an image for container",
 			)
 		}
-		if imageParts[1] == "latest" {
+
+		if imageParts[0] != defaultRegistry {
 			return errors.NewLintRuleError("CONTAINER003",
 				object.Identity()+"; container = "+c.Name,
 				nil,
-				"Image tag \"latest\" used",
+				"Image name should be "+defaultRegistry,
 			)
 		}
+
+		r := regexp.MustCompile("[A-Fa-f0-9]{64}")
+		switch os.Getenv("CI") {
+		case "true":
+			matched := r.MatchString(imageParts[1])
+			if !matched {
+				return errors.NewLintRuleError("CONTAINER003",
+					object.Identity()+"; container = "+c.Name,
+					nil,
+					"Image tag should be in form sha256 hash",
+				)
+			}
+		default:
+			if imageParts[1] != "imageHash" {
+				return errors.NewLintRuleError("CONTAINER003",
```

Comment:
I propose make unique rule identifiers to search in code in case of validation errors.

```suggestion
				return errors.NewLintRuleError("CONTAINER004",
```